src/test/cli.up.test.ts

```diff
@@ -44,6 +45,25 @@ describe('Dev Containers CLI', function () {
 			await shellExec(`docker rm -f ${containerId}`);
 		});
 
+		it('should execute successfully with valid config and dotfiles with secrets', async () => {
+			const testFolder = `${__dirname}/configs`;
+			let containerId: string | null = null;
+			await shellExec(`rm -f ${testFolder}/*.testMarker`, undefined, undefined, true);
+			const secrets = {
+				'SECRET1': 'SecretValue1',
+			};
+			await shellExec(`printf '${JSON.stringify(secrets)}' > ${testFolder}/test-secrets-temp.json`, undefined, undefined, true);
+
+			const res = await shellExec(`${cli} up --workspace-folder ${__dirname}/configs/image-with-git-feature --dotfiles-repository https://github.com/codspace/test-dotfiles --secrets-file ${testFolder}/test-secrets-temp.json`);
```

Comment:
Could we perhaps update the test repo to ensure we're passing these variables properly?

Reply:
This test is very basic. It only tests dotfiles still work with secrets file.
May be it is a good idea to make the dotfiles fail if expected secrets are not passed
Let me check

Reply:
Test added in https://github.com/devcontainers/cli/pull/544/commits/55ba3f86f2e329e16d75bbc7fc75d4e057d9218f